server/jira_to_md.go

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type jiraReplacer struct {
+	Type           string
+	RegExp         string
+	Replace        string
+	ReplaceStrFunc func(string) string // Takes priority
+}
+
+var (
+	jiraReplacers = []jiraReplacer{
+
+		{
+			Type:           "Lists",
+			RegExp:         `[ \t]*(#+)\s+`,
```

Comment:
JIRA sends the string as a single line.  Since jira2md is not currently splitting it into lines the ```^``` doesn't work as it's well into the single line string.

Reply:
I think you meant "JIRA sends the string as a single _*string*_" - in your own test you have multi-line input, with Windows-style `\r\n` line separation. AFAIU `^` matches the 0 position in the string, and every position immediately following `[\n]+`. I may be wrong on the specifics, but my point is that a string like "some text with a # hash is not a list" must not match, whereas "some text with a \r\n # list" must match. Right?